x/eibc/types/demand_order.go

```diff
@@ -32,6 +32,25 @@ func NewDemandOrder(rollappPacket commontypes.RollappPacket, price, fee math.Int
 }
 
 func (m *DemandOrder) ValidateBasic() error {
+	if len(m.Price) > 1 || len(m.Fee) > 1 {
+		return ErrMultipleDenoms
+	}
```

Comment:
Why do we even have a validate basic function if the data always comes from inside the system?
it should be correct by construction

Reply:
it shouold be correct, the enforcement done on the `Validate` instead of the `NewDemandOrder`
it probably should be used in case of loading orders on genesis as well